rust/azure_iot_operations_mqtt/tests/mock_connection_management.rs

```diff
@@ -573,3 +573,169 @@ async fn force_exit_while_disconnected() {
     let e = run_f.await.unwrap().unwrap_err();
     assert!(matches!(e.kind(), SessionErrorKind::ForceExit));
 }
+
+/// This test validates that a force exit can be done while a reauthentication is pending.
+#[tokio::test] // NOTE: This test WILL take > 20 seconds.
+async fn force_exit_during_reauth() {
+    let (connection_settings, session, mock_server, _, mock_sat_file) =
+        quick_setup_enhanced_sat_auth("test-force-exit-during-reauth-client");
+    let exit_handle = session.create_exit_handle();
+    let monitor = session.create_session_monitor();
+    assert!(!monitor.is_connected());
+
+    // Start the session run loop
+    let run_f = tokio::task::spawn(session.run());
+
+    // Validate that the CONNECT packet contains the expected values
+    let connect = mock_server.expect_connect_and_accept(true).await;
+    assert_eq!(
+        connect,
+        expected_connect_from_connection_settings(&connection_settings, false)
+    );
+
+    // Wait for connection to be established by Session in response to CONNACK
+    monitor.connected().await;
+
+    // Trigger reauthentication by updating the SAT file contents.
+    mock_sat_file.update_contents();
+
+    // Receive the AUTH packet on the server, but do not respond.
+    let auth = mock_server.expect_auth().await;
+    assert_eq!(auth, expected_reauth_from_sat_file(&mock_sat_file));
+
+    // Force exit while reauth is pending (i.e. have not responded with the server)
+    let graceful = exit_handle.force_exit();
+    assert!(graceful);
+
+    // Session was ended gracefully with DISCONNECT, and exited cleanly
+    let disconnect = mock_server.expect_disconnect().await;
+    assert_eq!(disconnect, session_end_disconnect());
+    monitor.disconnected().await;
+    assert!(run_f.await.unwrap().is_ok());
+
+    // Updating the SAT file after exit does not cause any AUTH packets to be sent
+    mock_sat_file.update_contents();
+    tokio::time::sleep(Duration::from_secs(11)).await;
+    mock_server.expect_no_packet();
+}
+
+/// Thist test validates that multiple SAT file updates does not cause multiple AUTH packets
+/// to be sent if done in rapid succession.
```

Comment:
why is this the case? is it because we wouldn't evaluate new sat file contents until the current auth handshake is complete and then go again with the latest?